air-quality-backend/src/database/mongo_db_operations.py

```diff
@@ -0,0 +1,45 @@
+from datetime import datetime
+import logging
+import os
+from pymongo import MongoClient, UpdateOne
+
+
+def get_collection(name: str):
+    uri = os.environ.get("MONGO_DB_URI")
+    db_name = os.environ.get("MONGO_DB_NAME")
+    client = MongoClient(uri)
+    return client[db_name][name]
+
+
+def upsert_data(collection_name: str, keys: list[str], data):
+    if len(data) == 0:
+        return
+    collection = get_collection(collection_name)
+    now = datetime.utcnow()
+
+    update_operations = [
+        UpdateOne(
+            {key: doc[key] for key in keys},
```

Comment:
These are the keys we will upsert on if there is a document already in the collection